### PR TITLE
Test Code

### PR DESCRIPTION
**Pull request by mercurrie**
Added Flowchart,
Added ASCII art
Added Comments to main.py

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+# function calculateScore takes in a list parameter and returns sum of list
+def calculateScore(list):
+    count = sum(list)
+
+    if count == 21:
+        return 21
+    elif count > 21:
+        for num in list:
+            if num == 11:
```
Comment (michaelhousman) [inline main.py:13]: This is good.  I thought initially that you were always converting 11 to 1 until I saw this condition.

File: main.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    return count
+
+# function compareScrore takes in user and dealer's scores, compares them, and returns a string
+def compareScore(userScore, dealerScore):
+  if userScore > 21 and dealerScore > 21:
+    return "You went over. You lose 😤"
+
+  if userScore == dealerScore:
+    return "Draw 🙃"
+  elif dealerScore == 0:
```
Comment (michaelhousman) [inline main.py]: You're checking for 0 to determine if the dealer won?

File: main.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    return "Lose, opponent has Blackjack 😱"
+  elif userScore == 0:
```
Comment (michaelhousman) [inline main.py]: Same as above.  Maybe I'm mis-interpreting things.

File: main.py [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+        for num in list:
+            if num == 11:
+                list.remove(num)
+                list.append(1)
+                count = sum(list)
+    
+    return count
+
+# function compareScrore takes in user and dealer's scores, compares them, and returns a string
+def compareScore(userScore, dealerScore):
```
Comment (michaelhousman) [inline main.py]: PEP-8 (look it up) conventions for Python variables and functions is all `snake_case`.  Please fix.

Comment (michaelhousman) [reply main.py]: You also need to type-cast all input variables like so:

def filter_object_fields(object: dict, included: List[str] = [], excluded: List[str] = []) -> dict:

It makes a world of difference when you're trying to debug.

File: main.py [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+    while dealerScore != 21 and dealerScore < 17:
+            dealer.append(dealCard())
+            dealerScore = calculateScore(dealer)
+        
+    print(f'\nYour final hand is: {user}, your final score is: {userScore}')
+    print(f"The dealer's final hand is: {dealer}, with a score of: {dealerScore}")
+    print(compareScore(userScore, dealerScore))
+    
+# call Blackjack function and clear console as long as User wants to continue playing
+while input("Would you like to play a game of Blackjack? Input 'y' for yes 'n' for no: ") == 'y':
```
Comment (michaelhousman) [inline main.py]: You need to call your script with something like this at the bottom:

if __name__ == '__main__':
    test_list = [i for i in range(0, 100)]
    print([i for i in shard_list(test_list, 5)])

**Review by michaelhousman — CHANGES_REQUESTED** [review]
1. Really good start.  I'm impressed.
2. Like how you used functions here. I saw that you didn't in some of the other repos and you should fix that
3. Added some comments that I would fix

File: main.py (hunk 2 above)
Comment (mercurrie) [reply main.py]: originally had 0 set as the value for Blackjack. I ended up just going with 21 for less confusion. This was an oversight on my part

File: main.py (hunk 5 above)
Comment (mercurrie) [reply main.py]: Great feedback, included in code

File: main.py (hunk 4 above)
Comment (mercurrie) [reply main.py]: I included the snake_case, and will continue with that format from now on.
As for the type-casting, I couldn't find any relevant resources or examples where it is used in functions like the example above. Hopefully you can explain this to me in more detail at a later time

Comment (michaelhousman) [reply main.py]: @mercurrie -- type-casting means that you identify the type (e.g., str, list) of the inputs and outputs like so:

def filter_object_fields(object, included, excluded):

VS:

def filter_object_fields(object: dict, included: List[str] = [], excluded: List[str] = []) -> dict:

What it does is allow the IDE to easily identify if there's any errors that might appear caused by incorrect type's before you run the code.  Use the function above as an example and see if you can specify what every input and output of every function should be.

**Comment by michaelhousman** [comment]
I know you merged already but I do want to see you fix that one issue that was outstanding from the last review.  Can you take out a PR with those changes?

**Pull request merged**